### PR TITLE
Make preview tabs work with all files

### PR DESCRIPTION
**Pull request by mnquintana**
Fixes https://github.com/atom/tabs/issues/157

This makes preview tabs work with any tab item that implements `::getPath()` instead of just instances of `TextEditor`.

@atom/feedback

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
:+1: Good call.

**Comment by mnquintana** [comment]
Do you think it's worth adding specs testing this with at least one filetype that won't open an editor (ie. an image)?

**Comment by lee-dohm** [comment]
I think so, yes. We don't want it accidentally breaking.

**Comment by mnquintana** [comment]
Alright, this should be ready for review.

**Comment by lee-dohm** [comment]
:+1: Looks good to me!

**Comment by kevinsawicki** [comment]
Yeah, I was deliberately defensive about this to start to prevent non-editor pane items from accidentally becoming preview tabs.

Say for example find and replace implemented a `getPath` method on their model class for some legitimate reason, it would now be a previewable item which seems less than ideal since that is something you'd want to stay open.

**Comment by lee-dohm** [comment]
Should we have a `supportsTemporaryView` attribute on models then? Rather than conflating whether they have a `getPath()` method and supporting temporary mode.

**Comment by mnquintana** [comment]
Hmm that's a good point, but this really should work with all files that Atom can open, instead of just text files. Another possibility is just adding more conditions to check for each possible file type that Atom can open, but that would be kinda brittle.

**Comment by mnquintana** [comment]
> Should we have a supportsTemporaryView attribute on models then? Rather than conflating whether they have a getPath() method and supporting temporary mode.

Oh yeah, that's a much better idea. Maybe `supportsPreviewTabs` though instead? (Just want to try to keep the terminology consistent.)

**Comment by kevinsawicki** [comment]
> Maybe supportsPreviewTabs though instead?

Maybe `isPreviewable`?

**Comment by lee-dohm** [comment]
Yeah ... though #6817 suggests making a "viewer" API and I feel like "viewer" and "preview" are too close in terms maybe?

**Comment by mnquintana** [comment]
@lee-dohm I was worried about that when I suggested the `preview` name for this feature, but I think they're distinct enough as long as we're consistent about it. (The only confusing thing being markdown-preview... hm.)

**Comment by lee-dohm** [comment]
I'll change the name of the suggested API to "viewer" then to be more distinct.

**Comment by kevinsawicki** [comment]
I'm okay leaving this using `getPath` to start and then we can then learn what packages this might impact  better and make it more opt-in in a second pass with a well-named API to be implemented by pane items.

Does that sound okay?

**Comment by lee-dohm** [comment]
Agreed :+1:

File: lib/tab-view.coffee [hunk 1]
```diff
@@ -5,7 +5,7 @@ path = require 'path'
 module.exports =
 class TabView extends HTMLElement
   initialize: (@item) ->
-    @isPreviewTab = atom.config.get('tabs.usePreviewTabs') and @item instanceof TextEditor
+    @isPreviewTab = atom.config.get('tabs.usePreviewTabs') and @item.getPath?
```
Comment (kevinsawicki) [inline lib/tab-view.coffee]: I'm going to update this to be `typeof @item.getPath is 'function'` just to be a little stricter.

**Comment by mnquintana** [comment]
Sounds good :+1:

Sorry for jumping the gun a bit on merging this :grimacing:

**Pull request merged**